comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -183,14 +183,49 @@ func (ra *remoteAgentRegistry) start() {
 				}
 
 				for _, id := range agentsToRemove {
-					delete(ra.agentMap, id)
-					log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					details, ok := ra.agentMap[id]
+					if ok {
+						if details.configStream != nil && details.configStream.stream != nil {
+							details.configStream.streamCancel()
+						}
+						delete(ra.agentMap, id)
+						log.Infof("Remote agent '%s' deregistered after being idle for %s.", id, remoteAgentIdleTimeout)
+					}
 				}
 
 				ra.agentMapMu.Unlock()
 			}
 		}
 	}()
+	go ra.sendConfigUpdatesLoop()
+}
+
+func (ra *remoteAgentRegistry) sendConfigUpdatesLoop() {
+	log.Info("Remote Agent config update loop started.")
+
+	for {
+
+		// TODO: Implement config update loop
```

Comment:
TODO: Actual implementation details regarding how/where we get the configuration updates is missing. `configsync`(?)